src/chain.cpp

```diff
@@ -58,6 +58,13 @@ const CBlockIndex *CChain::FindFork(const CBlockIndex *pindex) const {
     return pindex;
 }
 
+CBlockIndex *CChain::FindLatestBefore(int64_t nTime) const {
+    CBlockIndex *pindex = Tip();
+    while (pindex && pindex->pprev && pindex->GetBlockTime() > nTime)
```

Comment:
@jonasschnelli what is the variance on timestamps?  Is ~144 enough (or to much)?